src/plone/api/portal.py

```diff
@@ -281,19 +309,64 @@ def get_registry_record(name=None):
 
 
 @required_parameters('name', 'value')
-def set_registry_record(name=None, value=None):
+def set_registry_record(name=None, value=None, interface=None):
     """Set a record value in the ``plone.app.registry``
 
     :param name: [required] Name of the record
     :type name: string
     :param value: [required] Value to set
     :type value: python primitive
+    :param interface: interface whose attributes are plone.app.registry
+        settings
+    :type interface: zope.interface.Interface
     :Example: :ref:`portal_set_registry_record_example`
     """
     if not isinstance(name, str):
         raise InvalidParameterError(u"The parameter 'name' has to be a string")
+
+    if interface is not None and not IInterface.providedBy(interface):
+        raise InvalidParameterError(
+            u"The interface parameter has to derive from "
+            u"zope.interface.Interface"
+        )
+
     registry = getUtility(IRegistry)
-    if isinstance(name, str):
+
+    if interface is not None:
+        records = registry.forInterface(interface)
```

Comment:
This appears to be an exact copy of the code above. Would it not be better to call `get_registry_record` here?